### ISSUE TITLE
rust-analyzer false positive: “cannot index into &mut [[char; 20]; 20]” for `field[y][x]`

### ISSUE THREAD

**Issue by GitClo** [rust-lang/rust-analyzer#22514]
**rust-analyzer version**: rust_analyzer 0.3.2921-standalone

**rustc version**: rustc 1.95.0 (59807616e 2026-04-14) (Arch Linux rust 1:1.95.0-1)

**editor or extension**: NVIM v0.12.2 (+NVchad)                                                                                                                                                                                
Build type: RelWithDebInfo                                                                                                                                                              
LuaJIT 2.1.1774896198                                                                                                                                                                

**relevant settings**: default

**repository link (if public, optional)**: (eg. [rust-analyzer](https://github.com/rust-lang/rust-analyzer))

**code snippet to reproduce**:
```rust
const GRID_SIZE: usize = 20;
type Field = [[char; GRID_SIZE]; GRID_SIZE];

fn set_tile(field: &mut Field, x: usize, y: usize, ch: char) {
    field[y][x] = ch;
}

fn main() {
    let mut f: Field = [['-'; GRID_SIZE]; GRID_SIZE];
    set_tile(&mut f, 1, 2, 'x');
}

```

rust-analyzer reports a type error on valid array indexing. In the function set_tile, it flags field[y][x] = ch; with “cannot index into a value of type &mut [[char; 20]; 20]”, even though the code compiles and cargo check succeeds with no errors. This happens in a fresh project with the minimal snippet above.

**Comment by ChayimFriedman2** [comment]
Cannot reproduce. Do you have rust-src installed?

**Comment by GitClo** [comment]
> Cannot reproduce. Do you have rust-src installed?

(No.) My bad. Sorry for wasting time, didn't know I needed that for full functionality.
